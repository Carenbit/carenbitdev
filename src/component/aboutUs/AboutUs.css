#aboutUsPage {
  padding: 50px 30px;
  text-align: center;
  margin-top: 3%;
}

.aboutTitle {
  display: flex;
  flex-direction: column;
  align-items: center;
}

.welcomeMessage {
  font-size: 2.5rem;
  font-weight: 700;
  line-height: 48px;
}

.description {
  font-size: 20px;
  font-weight: 400;
  line-height: 32px;
  letter-spacing: -0.008em;
  text-align: center;
  max-width: 75%;
}

.blockCard {
  background-color: #fff9f0;
  padding: 50px 30px;
  margin: 20px;
  border-radius: 20px;
  margin-top: 3%;
  margin-bottom: 10%;
  display: flex;
  gap: 90px;
}

.blockCard:hover {
  background-color: #fff4f0;
  transition: 0.5s;
}

.reverse {
  flex-direction: row-reverse;
}

.statements {
  padding: 20px;
}

.statements h2 {
  font-size: 24px;
  font-weight: 600;
  line-height: 28.8px;
  letter-spacing: -0.02em;
  padding-bottom: 20px;
}

.statements p {
  font-size: 20px;
  font-weight: 400;
  line-height: 27px;
  letter-spacing: -0.02em;
  text-align: center;
}

@media only screen and (max-device-width: 1024px) {
  .blockCard {
    gap: 0px;
  }

  .description {
    font-size: 16px;
  }

  .statements p {
    font-size: 16px;
  }
  .statements {
    padding: 0;
  }

  .cardImage {
    height: 100%;
  }
}
@media only screen and (max-device-width: 840px) {
  .blockCard {
    flex-direction: column;
    gap: 0px;
  }
  .description {
    max-width: 100%;
  }
}
@media only screen and (max-device-width: 464px) {
  .welcomeMessage {
    font-size: 24px;
  }
  .blockCard {
    margin: 20px 0;
    padding: 3px;
  }
  .statements {
    padding: 0 important;
  }
  .statements p {
    font-size: 14px;
  }
}

@media only screen and (min-device-width: 1024px) {
  .blockCard {
    padding: 50px 10px;
  }
  .cardImage {
    height: 100%;
  }
  .statements {
    padding: 0;
  }
  .statements p {
    font-size: 18px;
  }
}
@media only screen and (min-device-width: 1200px) {
  .blockCard {
    margin: 2% 15%;
  }
}