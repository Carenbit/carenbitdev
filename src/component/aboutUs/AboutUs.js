import React from "react";
import "./AboutUs.css";
import mission from "../../assets/mission.png";
import vision from "../../assets/vision.png";

const AboutUs = (props) => {
  const { innerwidth } = props;

  return (
    <div id="aboutUsPage">
      <div className="aboutTitle">
        <h1 className="welcomeMessage">Welcome To Carenbit</h1>
        {innerwidth > 464 ? (
          <p className="description my-4 ">
            Carenbit Technology Services, founded in 2022, is a pioneering
            startup based in Pune. We specialize in providing expert solutions
            in Teamcenter PLM, aiming to unlock the full potential of businesses
            across diverse industries.
          </p>
        ) : (
          <p className="description my-4 ">
            At Carenbit, we're driven by a singular passion: to empower
            businesses with cutting-edge Product Lifecycle Management (PLM)
            solutions that redefine excellence.
          </p>
        )}
      </div>
      <div className="blockCard d-flex align-content-center justify-content-center">
        <img src={vision} alt="..." class="cardImage bg-white rounded-3 p-3" />
        <div className="statements px-1">
          <h2>Vision</h2>
          <p>
            Our vision is to pioneer the Teamcenter PLM landscape, providing
            clients with high-quality, innovative, and cost-effective solutions,
            ultimately paving the path for their triumph.
          </p>
        </div>
      </div>
      <div className="blockCard reverse">
        <img
          src={mission}
          alt="..."
          class="cardImage img-fluid  bg-white rounded-3 p-3"
        />
        <div className="statements">
          <h2>Mission</h2>
          <p>
            At Carenbit, we propel our clients to unparalleled success in
            product development by delivering tailor-made and comprehensive
            Teamcenter PLM solutions, empowering businesses to thrive and excel
            in the dynamic landscape of todayâ€™s evolving market.
          </p>
        </div>
      </div>
    </div>
  );
};

export default AboutUs;
