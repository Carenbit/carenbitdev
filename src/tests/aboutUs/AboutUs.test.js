import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import AboutUs from "../../component/aboutUs/AboutUs.js";

test("should have a heading with text 'Welcome To Carenbit'", () => {
  render(<AboutUs />);
  const heading = screen.getByText(/Welcome To Carenbit/i);
  expect(heading).toBeInTheDocument();
});

test("should have a heading with text 'Vision'", () => {
  render(<AboutUs />);
  const heading = screen.getAllByText(/Vision/i);
  expect(heading).toHaveLength(2);
});

test("should have a heading with text 'Mission'", () => {
  render(<AboutUs />);
  const heading = screen.getByText(/Mission/i);
  expect(heading).toBeInTheDocument();
});

test("should match screenshot of AboutUs component", () => {
  const { container } = render(<AboutUs />);
  expect(container).toMatchSnapshot();
});
