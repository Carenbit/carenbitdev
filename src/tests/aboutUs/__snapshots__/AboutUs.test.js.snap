// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should match screenshot of AboutUs component 1`] = `
<div>
  <div
    id="aboutUsPage"
  >
    <div
      class="aboutTitle"
    >
      <h1
        class="welcomeMessage"
      >
        Welcome To Carenbit
      </h1>
      <p
        class="description my-4 "
      >
        At Carenbit, we're driven by a singular passion: to empower businesses with cutting-edge Product Lifecycle Management (PLM) solutions that redefine excellence.
      </p>
    </div>
    <div
      class="blockCard d-flex align-content-center justify-content-center"
    >
      <img
        alt="..."
        class="cardImage bg-white rounded-3 p-3"
        src="vision.png"
      />
      <div
        class="statements px-1"
      >
        <h2>
          Vision
        </h2>
        <p>
          Our vision is to pioneer the Teamcenter PLM landscape, providing clients with high-quality, innovative, and cost-effective solutions, ultimately paving the path for their triumph.
        </p>
      </div>
    </div>
    <div
      class="blockCard reverse"
    >
      <img
        alt="..."
        class="cardImage img-fluid  bg-white rounded-3 p-3"
        src="mission.png"
      />
      <div
        class="statements"
      >
        <h2>
          Mission
        </h2>
        <p>
          At Carenbit, we propel our clients to unparalleled success in product development by delivering tailor-made and comprehensive Teamcenter PLM solutions, empowering businesses to thrive and excel in the dynamic landscape of todayâ€™s evolving market.
        </p>
      </div>
    </div>
  </div>
</div>
`;
